src/streaming/controllers/ListMpdController.js

```diff
@@ -0,0 +1,173 @@
+
+    let instance,
+        settings,
+        linkedPeriodList,
+        dashAdapter,
+        currentManifest,
+        manifestLoader,
+        mpdHasDuration
+
+    function setConfig(config) {
+        if (!config) {
+            return;
+        }
+
+        if (config.settings) {
+            settings = config.settings;
+        }
+
+        if (config.dashAdapter) {
+            dashAdapter = config.dashAdapter;
+        }
+
+        if (config.manifestLoader) {
+            manifestLoader = config.manifestLoader;
+        }
+    }
+
+    function initialize() {
+        eventBus.on(Events.IMPORTED_MPDS_LOADED, _onLinkedPeriodsLoaded, instance);
+        eventBus.on(MediaPlayerEvents.PLAYBACK_TIME_UPDATED, _triggerLoadImportMpd, instance);
+    }
+
+    function loadListMpdManifest(time) {
+        linkedPeriodList.forEach(linkedPeriod => {
+            if (_shouldLoadLinkedPeriod(linkedPeriod, time)) {
+                loadLinkedPeriod(currentManifest, linkedPeriod);
+            }
+        });
+    }
+
+    function _onLinkedPeriodsLoaded({ manifest, linkedPeriods }) {
+        currentManifest = manifest;
+        linkedPeriodList = linkedPeriods;
+
+        manifest.Period[0].start = manifest.Period[0].start ?? 0;
+        if (manifest.Period[0].start !== 0) {
+            throw new Error('The first period in a list MPD must have start time equal to 0');
+        } 
+
+        mpdHasDuration = manifest.hasOwnProperty(DashConstants.MEDIA_PRESENTATION_DURATION);
+        if (!mpdHasDuration) {
+            manifest.mediaPresentationDuration = 0;
+            for (let i = manifest.Period.length - 1; i >= 0; i--) {
+                manifest.mediaPresentationDuration += manifest.Period[i].duration;
+                if (manifest.Period[i].start) {
+                    manifest.mediaPresentationDuration += manifest.Period[i].start;
+                    break;
+                }
+            }
+        }
+
+        const startPeriod = linkedPeriodList.find(period => period.start === 0);
+        if (startPeriod) {
+            loadLinkedPeriod(manifest, startPeriod);
+        } else {
+            eventBus.trigger(Events.MANIFEST_UPDATED, { manifest: manifest });
+        }
+    }
+
+    function loadLinkedPeriod(manifest, period) {
+        const relativePath = period.ImportedMPD.uri; 
+        const baseUrl = period.BaseURL ?? manifest.BaseURL;
+        const resolvedUri = baseUrl ? baseUrl[0].__text + relativePath : relativePath;
+
+        const updatedManifest = new Promise(resolve => {
+            manifestLoader.load(resolvedUri, null, null, true)
+                .then((importedManifest) => {
+                    dashAdapter.mergeManifests(manifest, importedManifest, period, mpdHasDuration);
+                }, () => {
+                    dashAdapter.mergeManifests(manifest, null, period, mpdHasDuration);
+                })
+                .then(() => {
+                    eventBus.trigger(Events.MANIFEST_UPDATED, { manifest });
```

Comment:
Is there a scenario in which multiple linked periods are resolved at the same time and this event is dispatched multiple times?

Reply:
Yes, that can happen when two periods have the same calculated value for `start time - earliestResolutionTimeOffset`.

For example:
- Period 1 has a start time of 10 and an earliest resolution time of 5.
- Period 2 has a start time of 15 and an earliest resolution time of 10